src/main/java/core/basesyntax/lib/Injector.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.lib;
+
+import core.basesyntax.dao.BetDao;
+import core.basesyntax.dao.UserDao;
+import core.basesyntax.factory.Factory;
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.function.Predicate;
+
+public class Injector implements Inject {
+    @Override
+    public Class<? extends Annotation> annotationType() {
+        return null;
+    }
+
+    public static Object getInstance(Class clazz) throws NoSuchMethodException,
+            IllegalAccessException, InvocationTargetException, InstantiationException {
+        Constructor constructor = clazz.getDeclaredConstructor();
+        Object instance = constructor.newInstance();
+        Field[] declaredFields = clazz.getDeclaredFields();
+        for (Field field: declaredFields) {
+            if (field.getAnnotation(Inject.class) != null) {
+                field.setAccessible(true);
+                Predicate predicate = o -> !(Factory.getBetDao().getClass()
+                        .isAnnotationPresent(Dao.class)
+                        || !(Factory.getUserDao().getClass()
+                        .isAnnotationPresent(Dao.class)));
+                if (predicate.test(field)) {
+                    throw new RuntimeException("No such class annotation");
+                }
+                if (field.getType() == UserDao.class) {
+                    field.set(instance, Factory.getUserDao());
+                } else if (field.getType() == BetDao.class) {
+                    field.set(instance, Factory.getBetDao());
+                }
```

Comment:
I think u misunderstood the task.
1. Find the fields of the class, which being injected, for `@Inject`
2. Check the `Type` of this fields for `@Dao`